### PR TITLE
constexpr updates

### LINKED ISSUE #1720: constexpr updates for C++17
Now that OTIO has updated to C++17, are there additional functions in OpenTime that we can mark as `constexpr`?

### PR DESCRIPTION
**Pull request by darbyjohnston**
Fixes #1720

Changes:
* Add `constexpr` to more functions in OpenTime
* Remove redundant `inline` on `constexpr` functions
* Move `constexpr` before the return type on functions

### CODE REVIEW

File: src/opentime/timeTransform.h [hunk 1]
```diff
@@ -29,9 +29,9 @@ class TimeTransform
     constexpr double rate() const noexcept { return _rate; }
 
     constexpr TimeTransform(TimeTransform const&) noexcept  = default;
-    TimeTransform& operator=(TimeTransform const&) noexcept = default;
+    constexpr TimeTransform& operator=(TimeTransform const&) noexcept = default;
```
Comment (meshula) [inline src/opentime/timeTransform.h:32]: does constexpr have a meaning here? It seems.... cosmetic?

Comment (meshula) [reply src/opentime/timeTransform.h]: yes, that's right.

**Comment by darbyjohnston** [comment]
The last build action failed due to a codecov error:
```
[2024-04-09T16:08:22.025Z] ['error'] There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 404 - {'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}
Error: Codecov: Failed to properly upload: The process '/home/runner/work/_actions/codecov/codecov-action/v3.1.1/dist/codecov' failed with exit code 255
```

**Review by meshula — APPROVED** [review]
lgtm

**Pull request merged**